layers/+emacs/tabs/funcs.el

```diff
@@ -0,0 +1,60 @@
+(defcustom spacemacs-tabs-auto-hide nil
+  "If non-nil hide tabs automatically after TABS-AUTO-HIDE-DELAY seconds."
+  :type '(boolean)
+  :group 'centaur-tabs)
+
+(defcustom spacemacs-tabs-auto-hide-delay 2
+  "Tabs auto hide delay in seconds."
+  :type '(float)
+  :group 'centaur-tabs)
+
+(defun spacemacs//tabs-timer-initialize (secs)
+  (setq spacemacs-tabs-timer (run-with-timer secs nil (lambda () (centaur-tabs-local-mode 1)))))
+
+(defun spacemacs//tabs-timer-hide ()
+  (spacemacs//tabs-timer-initialize spacemacs-tabs-auto-hide-delay))
+
+(when spacemacs-tabs-auto-hide
+  (add-hook 'window-setup-hook 'spacemacs//tabs-timer-hide)
+  (add-hook 'find-file-hook 'spacemacs//tabs-timer-hide)
+  (add-hook 'change-major-mode-hook 'spacemacs//tabs-timer-hide))
+
```

Comment:
No batch processing in funcs files please. This needs to be wrapped into a defun or into a specific init function. Otherwise it will cause side effects during the multi-phase loading process which expects to only find side effect free function definitions.